packages/toolpad-core/src/SignInPage/SignInPage.tsx

```diff
@@ -40,6 +39,29 @@ import FusionAuthIcon from './icons/FusionAuth';
 import { BrandingContext, RouterContext } from '../shared/context';
 import { DocsContext } from '../internal/context';
 
+const getCommonTextFieldProps = (theme: Theme, baseProps: TextFieldProps = {}): TextFieldProps => ({
+  required: true,
+  fullWidth: true,
+  ...baseProps,
+  slotProps: {
+    ...baseProps.slotProps,
+    htmlInput: {
+      sx: {
+        paddingTop: theme.spacing(1),
+        paddingBottom: theme.spacing(1),
+      },
+      ...baseProps.slotProps?.htmlInput,
+    },
+    inputLabel: {
+      sx: {
+        lineHeight: theme.typography.pxToRem(12),
+        fontSize: theme.typography.pxToRem(14),
+      },
+      ...baseProps.slotProps?.inputLabel,
+    },
```

Comment:
```suggestion
    htmlInput: {
      ...baseProps.slotProps?.htmlInput,
      sx: [
        {
          paddingTop: theme.spacing(1),
          paddingBottom: theme.spacing(1),
        },
        ...(Array.isArray(baseProps.slotProps?.htmlInput?.sx) ? baseProps.slotProps?.htmlInput?.sx : [baseProps.slotProps?.htmlInput?.sx      
    },
    inputLabel: {
      ...baseProps.slotProps?.inputLabel,
      sx: [
        {
          lineHeight: theme.typography.pxToRem(12),
          fontSize: theme.typography.pxToRem(14),
        },
        ...(Array.isArray(baseProps.slotProps?.inputLabel?.sx) ? baseProps.slotProps?.inputLabel?.sx : [baseProps.slotProps?.inputLabel?.sx
        ],
      },
    },
```

You can extract the sx prop into variables so we don't repeat the expressions.